### ISSUE TITLE
``If-None-Match: *`` should not raise a 412 on a deleted record.

### ISSUE THREAD

**Issue by Natim** [mozilla-services/cliquet#365]
Today, we are using ``If-None-Match: *`` to create the record if it doesn't exists yet.

The thing is, if the record have been deleted, then ``If-None-Match: *`` fails because the record exists as deleted.

It might be a feature also it means we should describe a way to create a record if it doesn't exists or have been deleted.

**Comment by Natim** [comment]
I think we should not consider deleted for the specific case of ``If-None-Match: *``

**Comment by Natim** [comment]
- Notification for deleted object makes sense :+1: 
 - Conflict handling with deleted items makes sense :+1: 
 - Conflict handling on creation with deleted items doesn't makes sense. :-1:

**Comment by Natim** [comment]
@michielbdejong do you have any insights for us here?

**Comment by leplatrem** [comment]
This was described in https://github.com/mozilla-services/cliquet/issues/249

**Comment by michielbdejong** [comment]
> Conflict handling on creation with deleted items doesn't makes sense.

I agree. An example of this situation would be:
* the resource does not exist
* client1 successfully creates the record, then successfully deletes it again, before client2 discovers it ever existed.
* client2 "thinks" the resource never existed, and tries to create it, possibly with the same content as what client1 put into it, possibly with different content.
* I agree client2's action should succeed in this case.

A way to see it is that the versioning of a resource should always be a linear step-by-step path, so that there is only one clear sequence of versions the resource goes through on the server, from its creation to its deletion. But once the resource has been deleted, this chain is forgotten by the server and it's back to its default state of non-existence.

At least that's the way we did it in remoteStorage.

The reasoning in #249 is different because that's based on modified-timestamps, not versions. There tombstones made more sense, maybe. But if you just look at ETags, then my intuition is to consider the lifespan of a resource from creation to deletion, and once it's gone, its lifespan is over and it can be forgotten.

**Comment by Natim** [comment]
Here as the steps to reproduce:

```
$ http PUT https://kinto.dev.mozaws.net/v1/buckets/default/collections/tasks/records/9055850d-3038-4979-b1dd-0e5084fbc0f3 --auth natim:
$ http DELETE https://kinto.dev.mozaws.net/v1/buckets/default/collections/tasks/records/9055850d-3038-4979-b1dd-0e5084fbc0f3 --auth natim:
$ http PUT https://kinto.dev.mozaws.net/v1/buckets/default/collections/tasks/records/9055850d-3038-4979-b1dd-0e5084fbc0f3 If-None-Match:* --auth natim:
```

**Comment by Natim** [comment]
This has been implemented in Cliquet 2.10.0
